relayer/tools/gen-mmr.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"fmt"
+	"hash"
+
+	"reflect"
+
+	"github.com/cbergoon/merkletree"
+	secp256k1 "github.com/ethereum/go-ethereum/crypto"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/sha3"
+)
+
+// Validator keypair (secp256k1)
+type Keypair struct {
+	publicBytes []byte
+	public      *ecdsa.PublicKey
+	private     *ecdsa.PrivateKey
+}
+
+type ValidatorPublicKey []byte
+type ParaBlockHash [32]byte
+
+// Uninterpreted hash type
+type H256 [32]byte
+
+// Leaf node for MMR
+type MMRLeaf struct {
+	blockHash        H256
+	paraHeadsRoot    H256
+	hasValidatorRoot bool
```

Comment:
I believe we can remove this bool as the nil contents of `validatorRoot` can indicate if the MMR has a validator root, correct?

Reply:
Well, `hasValidatorRoot` is still useful for SCALE-encoding the leaf. The encoder/decoder can use the flag to encode/decode appropriately.